### PR TITLE
Functional test seeding, Lucene PackageId search, and CI test gating

### PR DESCRIPTION
**Pull request by joelverhagen**
## Summary

Extends the public CI pipeline with functional test execution, adds test data seeding infrastructure, and improves Lucene search to support `packageid:` exact-match queries.

### CI Pipeline & Test Infrastructure
- Consolidate Aspire CI steps into a single pipeline task with try/finally for reliable cleanup
- Add `Seed-FunctionalTestData.ps1` script that runs `seedfunctionaltests` to create users, orgs, API keys, and push test packages via HTTP
- Simplify `Start-AspireHost.ps1` and `Stop-AspireHost.ps1` (use explicit PID parameter instead of PID file)
- Run P0, P1, and P2 functional tests with `--blame-hang-timeout 600s`

### GalleryTools Commands
- Extract shared `GalleryOperations` class from duplicated code across `CreateUserCommand`, `CreateOrganizationCommand`, `CreateApiKeyCommand`, and `SeedFunctionalTestsCommand`
- Add `seedfunctionaltests` command: creates test accounts, orgs, API keys, pushes test packages via HTTP API, and outputs `settings.CI.json`
- Add `seedsinglepackage` command: lightweight single-package seed used by AppHost (creates user + org + API key + HTTP push in one invocation)
- Add `createapikey` and `createorganization` commands

### AppHost
- Replace 3 separate seed resources (seed-user, seed-org, seed-package) with single `seedsinglepackage` resource
- Add `db2catalog` dependency on `seedsinglepackage` so V3 pipeline processes seeded data
- Add test data nupkg files for functional tests

### Lucene Search
- Add `packageid:` field support to `LuceneSearchService` for exact-match package ID queries
- Add DB fallback for `packageid:` queries in `SearchAdaptor` when Lucene is unavailable

### Test Gating
- Add `NeedsManyVersionsFact`, `NeedsSearchServiceTheory`, `NeedsStatisticsServiceFact/Theory` attributes to skip tests when CI environment lacks those capabilities
- Move error handling tests (HTTP->HTTPS redirect, simulated errors) to `AppServiceTests` category (require IIS custom error pages)
- Move `DuplicatePushesAreRejectedAndNotDeleted` to `AppServiceTests` (160 concurrent pushes cause thread pool starvation in CI)
- Add `HasManyVersions`, `HasSearchService`, `HasStatisticsService` config flags

### Unit Tests
- Add `SearchWithPackageIdReturnsExactMatchOnly` Lucene test
- Add `PackageId`/`packageid` inline data to existing exact-field search test

## Test Results

CI build #1403995: **165 passed, 0 failed, 10 skipped**

### CODE REVIEW

File: .pipelines/NuGetGallery-CI-dnceng.yml [hunk 1]
```diff
@@ -223,21 +223,72 @@ stages:
...
               workingFolder: $(NuGetGalleryPath)
-          - task: PowerShell@1
-            name: start_aspire
-            displayName: Start Aspire Host
+          - task: PowerShell@2
+            name: functional_tests
+            displayName: Aspire Functional Tests
+            timeoutInMinutes: 30
+            env:
+              ConfigurationFilePath: $(NuGetGalleryPath)\tests\NuGetGallery.FunctionalTests\settings.CI.json
```
Comment (japarson) [inline .pipelines/NuGetGallery-CI-dnceng.yml:231]: I can't find this file in this PR or in the main branch, but maybe I'm doing something wrong.

Comment (japarson) [reply .pipelines/NuGetGallery-CI-dnceng.yml:231]: Looks like it is generated by the seed data script. My bad.

File: src/GalleryTools/Commands/CreateApiKeyCommand.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using Autofac;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Services.Entities;
+using NuGetGallery;
+using NuGetGallery.Authentication;
+using NuGetGallery.Infrastructure.Authentication;
+
+namespace GalleryTools.Commands
+{
+	public static class CreateApiKeyCommand
```
Comment (japarson) [inline src/GalleryTools/Commands/CreateApiKeyCommand.cs]: Are we supposed to be using tabs? I usually see spaces, but I also see in some of the edited files in this PR that tabs are being used in some files.

File: src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class LuceneSearchService : ISearchService
     {
         private readonly Lucene.Net.Store.Directory _directory;
 
-        private static readonly string[] FieldAliases = new[] { "Id", "Title", "Tag", "Tags", "Description", "Author", "Authors", "Owner", "Owners" };
```
Comment (japarson) [inline src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs:25]: What's the difference between Id and PackageId? Why is PackageId being added? Is this a new functionality separate from the CI functional test extension?

File: tests/NuGetGallery.FunctionalTests/ErrorHandling/ErrorHandlingTests.cs [hunk 4]
```diff
@@ -82,7 +82,7 @@ public async Task RejectedUrl(string relativePath)
         /// </summary>
         [Theory]
         [Priority(2)]
-        [Category("P2Tests")]
```
Comment (japarson) [inline tests/NuGetGallery.FunctionalTests/ErrorHandling/ErrorHandlingTests.cs:85]: What exactly is an app service test and when are we running it? Do these functional tests actually spin up an app service and test it in the PR build like how we do with the staging app service when deploying NuGetGallery?

**Review by japarson — DISMISSED** [review]

File: .pipelines/NuGetGallery-CI-dnceng.yml (hunk 1 above)
Comment (joelverhagen) [reply .pipelines/NuGetGallery-CI-dnceng.yml:231]: Yup, generated!

File: src/GalleryTools/Commands/CreateApiKeyCommand.cs (hunk 2 above)
Comment (joelverhagen) [reply src/GalleryTools/Commands/CreateApiKeyCommand.cs]: Definitely spaces, this is my (AI) mistake. Will fix!

File: src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs (hunk 3 above)
Comment (joelverhagen) [reply src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs:25]: The semantics are shared with the Azure Search service `id:` is a tokenized match against package ID. `packageid:` is exact match. We needed this to make the functional tests pass, one of which uses `packageid:`. They usually run against the real Azure Search service which implements this syntax.

File: tests/NuGetGallery.FunctionalTests/ErrorHandling/ErrorHandlingTests.cs (hunk 4 above)
Comment (joelverhagen) [reply tests/NuGetGallery.FunctionalTests/ErrorHandling/ErrorHandlingTests.cs:85]: App Service is the Azure service that runs NuGetGallery.dll after deployment. 

The functional tests run against some endpoint, in this case https://localhost (NuGetGallery running in the CI via IIS express).

The "AppService" category tests are run only against a deployed App Service. In this case, the error handling tests only work against real App Service because there is some error handling edge cases that behave differently locally vs. on App Service and these tests capture the latter.

**Review by japarson — APPROVED** [review]
